src/main/java/com/it4045/pokemonviewer/enterprise/dto/Pokemon.java

```diff
@@ -2,35 +2,48 @@
 
 import lombok.Data;
 
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 
 public @Data
 class Pokemon {
 
-    private int pokemonId;
-    private String pokemonNumber;
-    private String pokemonName;
-    private String pokemonImg;
-    private List<String> pokemonType;
-    private String pokemonHeight;
-    private String pokemonWeight;
-    private String pokemonCandy;
-    private String pokemonEgg;
-    private double pokemonSpawnChoice;
-    private double pokemonAvgSpawns;
-    private String pokemonSpawnTime;
-    private List<Double> pokemonMultipliers;
-    private List<String> pokemonWeakness;
-    private List<Evolution> pokemonNextEvolution;
-    private List<Evolution> pokemonPrevEvolution;
+    private int id;
+    private String number;
+    private String name;
+    private String img;
+    private List<String> type;
+    private String height;
+    private String weight;
+    private String candy;
+    private String egg;
+    private double spawnChoice;
+    private double avgSpawns;
+    private String spawnTime;
+    private List<Double> multipliers;
+    private List<String> weakness;
+    private List<Map.Entry<Pokemon, String>> previousEvolution;
+    private List<Map.Entry<Pokemon, String>> nextEvolution;
 
     public Pokemon(){ }
 
-    public Pokemon(String pokemonName, String image, List<String> weakness, List<String> type, String spawnTime) {
-        this.pokemonName = pokemonName;
-        this.pokemonImg = image;
-        this.pokemonWeakness = weakness;
-        this.pokemonType = type;
-        this.pokemonSpawnTime =spawnTime;
+    public Pokemon(String name, String image, List<String> weakness, List<String> type, String spawnTime) {
+        this.name = name;
+        this.img = image;
+        this.weakness = weakness;
+        this.type = type;
+        this.spawnTime =spawnTime;
     }
-}
+
+    public Pokemon(String name, String image, List<Map.Entry<Pokemon, String>> previousEvolution, List<Map.Entry<Pokemon, String>> nextEvolution) {
+        this.name = name;
+        this.img = image;
+        this.weakness = weakness;
+        this.type = type;
+        this.spawnTime =spawnTime;
+
+        this.previousEvolution = previousEvolution;
+        this.nextEvolution = nextEvolution;
+    }
+}
```

Comment:
Third suggestion: add a new constructor to Pokemon that includes evolution data.

Since the Evolution class has been incorporated into the Pokemon class, it is possible to supply a Pokemon's evolution data at the same time that it is being created. For this, I added another constructor. There may be times when a Pokemon's evolution data is not yet known, so I left the other two constructors that had already been created, but the new one should be helpful if you have all of the data that you need from the JSON API.